RevenueCat/Plugins/Android/PurchasesWrapper.java

```diff
@@ -395,6 +406,29 @@ private static void sendErrorPurchase(ErrorContainer errorContainer) {
         sendJSONObject(jsonObject, MAKE_PURCHASE);
     }
 
+    @NonNull
+    private static OnResult getLogInListener(final String method) {
+        return new OnResult() {
+            @Override
+            public void onReceived(Map<String, ?> map) {
+                JSONObject jsonObject = new JSONObject();
+                try {
+                    Map<String, ?> purchaserInfoMap = (Map<String, ?>)map.get("purchaserInfo");
+                    jsonObject.put("purchaserInfo", MappersHelpersKt.convertToJson(purchaserInfoMap));
+                    jsonObject.put("created", (Boolean)map.get("created"));
```

Comment:
since we control the data types, the casting here should be safe. 

However, let me know if there's a better way to do this

Reply:
i think it should be fine, if we wanted to be really safe we could catch an InvalidCastException 🤷‍♀️

Reply:
good call, added